### PR TITLE
Add macOS CI build with codesigning etc.

### PR DESCRIPTION
**Pull request by adriweb**
Tested the generated artifact, works fine!

### CODE REVIEW

File: .github/workflows/macOS.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        os: [macOS-latest]
+
+    steps:
+    - name: Checkout Git Repo
+      uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+
+    - name: Install Qt
+      uses: jurplel/install-qt-action@v3
```
Comment (Vogtinator) [inline .github/workflows/macOS.yml:27]: Please pin third-party actions by full sha256

File: .github/workflows/macOS.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+name: Build macOS
+
+on:
+  push:
+    branches: [ master, ci/macOS ]
+  pull_request:
```
Comment (Vogtinator) [inline .github/workflows/macOS.yml:6]: Do PR builds also have access to the secrets?

Comment (adriweb) [reply .github/workflows/macOS.yml:6]: Only if they are from this repo (so, PR from one branch to another), considering I added the secrets to this repo (not the org)

Comment (Vogtinator) [reply .github/workflows/macOS.yml:6]: Perfect

**Pull request merged**